test/src/unittests/tonal/test_tonicindianartmusic.py

```diff
@@ -33,66 +34,88 @@ def testInvalidParam(self):
         self.assertConfigureFails(TonicIndianArtMusic(), { 'harmonicWeight': 1 })
         self.assertConfigureFails(TonicIndianArtMusic(), { 'hopSize': -1 })
         self.assertConfigureFails(TonicIndianArtMusic(), { 'magnitudeCompression': -1 })
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'magnitudeCompression': 2 })        
         self.assertConfigureFails(TonicIndianArtMusic(), { 'magnitudeThreshold': -1 })
         self.assertConfigureFails(TonicIndianArtMusic(), { 'maxTonicFrequency': -1 })
         self.assertConfigureFails(TonicIndianArtMusic(), { 'minTonicFrequency': -1 })
         self.assertConfigureFails(TonicIndianArtMusic(), { 'numberHarmonics': 0 })
         self.assertConfigureFails(TonicIndianArtMusic(), { 'numberHarmonics': -1 })        
         self.assertConfigureFails(TonicIndianArtMusic(), { 'numberSaliencePeaks': 0})
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'numberSaliencePeaks': 16})        
         self.assertConfigureFails(TonicIndianArtMusic(), { 'referenceFrequency': -1 })
         self.assertConfigureFails(TonicIndianArtMusic(), { 'sampleRate':   -1 })
 
-    def testZeros(self):
-        self.assertRaises(RuntimeError, lambda: TonicIndianArtMusic()(zeros(1024)))   
+    def testEmpty(self):
+        self.assertRaises(RuntimeError, lambda: TonicIndianArtMusic()([]))   
 
-    def testOnes(self):
-        referenceTonic =108.86099243164062     
-        tonic   = TonicIndianArtMusic()(ones(1024))
-        self.assertAlmostEqual(referenceTonic, tonic, 8)
+    def testSilence(self):
+        silence = np.zeros(int(np.abs(np.random.randn()) * 30. * 44100))
+        self.assertRaises(RuntimeError, lambda: TonicIndianArtMusic()(silence))
 
-    def testNegativeInput(self):
-        tonic = TonicIndianArtMusic()([-1]*1024)
-        referenceTonic =108.86099243164062     
-        tonic   = TonicIndianArtMusic()([-1]*1024)
-        self.assertAlmostEqual(referenceTonic, tonic, 8)
+    def testOnes(self):
+        # FIXME: The tonic value 117.36 is not returned everytime. Sometimes 55 is returned.
```

Comment:
What is the source for deciding the correct `referenceTonic` value? The constant input signal doesn't contain a workable frequency spectrum, so i don't see the point of this test.

Reply:
use ground truth from data set. and check its OK without silences